ddtrace/llmobs/_llmobs.py

```diff
@@ -75,30 +92,127 @@ def _stop_service(self) -> None:
             log.warning("Failed to shutdown tracer", exc_info=True)
 
     @classmethod
-    def enable(cls, tracer=None):
+    def _toggle(cls, switch: bool) -> None:
+        # _toggle is a helper function to ensure the state of config._llmobs_enabled is in sync with the class attribute
+        # since our integrations depend on config._llmobs_enabled to set llmobs data
+        config._llmobs_enabled = switch
+        cls.enabled = switch
+
+    @classmethod
+    def enable(
+        cls,
+        ml_app: Optional[str] = None,
+        integrations: Optional[List[str]] = None,
+        dd_llmobs_no_apm: bool = False,
+        dd_site: Optional[str] = None,
+        dd_api_key: Optional[str] = None,
+        dd_env: Optional[str] = None,
+        dd_service: Optional[str] = None,
+        tracer=None,
+    ):
+        """
+        Enable LLM Observability tracing.
+
+        :param str ml_app: The name of your ml application.
+        :param List[str] integrations: A list of integrations to enable auto-tracing for.
+        :param bool dd_llmobs_no_apm: Set to `true` to disable sending non-LLM Observability data to Datadog.
+        :param str dd_site: Your datadog site.
+        :param str dd_api_key: Your datadog api key.
+        :param str dd_env: Your environment name.
+        :param str dd_service: Your service name.
+        """
         if cls._instance is not None:
             log.debug("%s already enabled", cls.__name__)
             return
 
+        if os.getenv("DD_LLMOBS_ENABLED") and not asbool(os.getenv("DD_LLMOBS_ENABLED")):
+            LLMObs._toggle(False)
+            log.debug("LLMObs.enable() called when DD_LLMOBS_ENABLED is set to false, not starting LLMObs service")
+            return
+
+        # grab required values for LLMObs
+        # reading from environment variables is needed in case preload script wasn't called.
+        config._dd_site = config._dd_site or os.getenv("DD_SITE") or dd_site
```

Comment:
for a future pr - i'm wondering if as we add more configuration options it'd be better to have our own config coalescing in a separate module, similar as to how it's done in `config.py`, instead of doing it all here. would make it a bit cleaner and probably easier to test too. wdyt?